src/services/socket/socket.gateway.ts

```diff
@@ -0,0 +1,60 @@
+import { Server, Client, Socket } from 'socket.io';
+import {
+  SubscribeMessage,
+  WebSocketGateway,
+  WsResponse,
+  OnGatewayInit,
+  OnGatewayConnection,
+  OnGatewayDisconnect,
+} from '@nestjs/websockets';
+import { Logger, OnModuleDestroy } from '@nestjs/common';
+
+@WebSocketGateway()
+export class SocketGateway
+  implements
+    OnGatewayInit,
+    OnGatewayConnection,
+    OnGatewayDisconnect,
+    OnModuleDestroy {
+  private server: Server;
+
+  constructor(private readonly logger: Logger) {}
+
+  afterInit(server: Server) {
+    this.server = server;
+  }
+
+  handleConnection(client: Socket) {
+    const { address } = client.handshake.query;
+    this.logger.log(`WebSocket: ${address} client connected`);
+
+    if (address) {
+      client.join(address); // join to user room
+
+      // TODO: find company of user and join to company room
+    }
+  }
+
+  handleDisconnect(client: Socket) {
+    this.logger.log(
+      `WebSocket: ${JSON.stringify(client.rooms)} client disconnected`,
+    );
+  }
+
+  onModuleDestroy() {
+    this.server.close();
+  }
+
+  @SubscribeMessage('message')
+  handleMessage(client: Client, payload: any): WsResponse<void> {
+    this.logger.log('WebSocket: got event message from client');
+    this.logger.log(client);
+    this.logger.log(payload);
+
+    return undefined;
```

Comment:
According to Nest.js docs we can use any "falsy" value here to not to respond to a user, so, probably it will be better to return `false` instead of `undefined`.
```suggestion
    return undefined;
```

Reply:
done https://github.com/sigmaledger/im-api/pull/2/commits/ff2dd05f95731adb1f1e4f0fed530077a7644b88